airflow/models/dagrun.py

```diff
@@ -494,12 +542,13 @@ def verify_integrity(self, session: Session = None):
                 session.add(ti)
 
         try:
-            session.commit()
+            session.flush()
```

Comment:
What's the reason for the integration error here? Is the Integrity error here still valid if we do not commit to the database? I think flush is not likely to generate Integrity Error at all.

Shoudl we start caring about isolation level of the transactions we have ? Will it work when we have different isolation levels set ?

Reply:
Not sure yet -- https://github.com/apache/airflow/pull/10136 was where this was added only recently.